packages/lesswrong/components/async/localStorageHandlers.js

```diff
@@ -0,0 +1,66 @@
+import ls from 'local-storage';
+
+export const commentLSHandlers = {
+  get: ({doc, name}) => {
+    if (doc && doc._id) { // When restoring the edit state for a specific comment, ask for permission
+      const savedState = ls.get(doc._id);
+      if (savedState && Meteor.isClient && window) {
+        const result = window.confirm("We've found a previously saved state for this comment, would you like to restore it?")
+        if (result) { return savedState }
+      } else {
+        return null;
+      }
+    } else if (doc && doc.parentCommentId) {
+      return ls.get('parent:' +  doc.parentCommentId)
+    } else if (doc && doc.postId) {
+      return ls.get('post:' + doc.postId)
+    } else {
+      return null
+    }
+  },
+  set: ({state, doc, name}) => {
+    if (doc && doc._id) {
+      ls.set(doc._id, state);
+    } else if (doc && doc.parentCommentId) {
+      ls.set('parent:' + doc.parentCommentId, state);
+    } else if (doc && doc.postId, state) {
+      ls.set('post:' + doc.postId, state);
+    }
+  },
+  reset: ({doc, name}) => {
+    if (doc._id) { ls.remove(doc._id) }
+    else { ls.remove(name) }
+  }
+}
+
+export const defaultLSHandlers = {
```

Comment:
Would be nicer if this and `commentLSHandlers` shared a base class, though it's not essential. The copy-pastiness is worrying since these can get out of sync.

Reply:
Yeah, but I don't like introducing classes when we basically don't use them anywhere else in any real form in the code base.